Script/WorldSelector.gd

```diff
@@ -20,3 +21,13 @@ func _on_extra_worlds_button_pressed() -> void:
 func _on_back_button_pressed() -> void:
 	ExtraWorldSelector.visible = false
 	MainWorldSelector.visible = true
+
+
+func _on_main_world_selector_settings_selected() -> void:
+	MainWorldSelector.visible = false
+	SettingsSelector.visible = true
+
+
+func _on_settings_selector_back() -> void:
+	SettingsSelector.visible = false
+	MainWorldSelector.visible = true
```

Comment:
You could, if you preferred, connect that signal to `_on_back_button_pressed` and hide both `ExtraWorldSelector` and `SettingsSelector` there.